### ISSUE TITLE
augmentation.RandomGaussianIllumination: center=0.5 misses the image centre on odd axes (round-half-even + even-window mean shift)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4811]
`RandomGaussianIllumination`'s `center` does not land where it says. `center=0.5` puts the peak on the image centre for a 5-, 9- or even-sized axis, but one pixel past it for a 3-, 7- or 11-sized axis.

```python
import torch, kornia.augmentation as K
for L in (3, 5, 7, 9, 11, 6, 8):
    a = K.RandomGaussianIllumination(gain=(0.5, 0.5), sigma=(0.2, 0.2), center=(0.5, 0.5), sign=(1., 1.), p=1.)
    torch.manual_seed(0); a(torch.zeros(1, 1, L, L)); w = a._params["gradient"][0, 0].sum(0)
    print(L, int(w.argmax()), float((w * torch.arange(L)).sum() / w.sum()), (L - 1) / 2)
# 3  2 1.795 1.0
# 5  2 2.000 2.0
# 7  4 3.892 3.0
# 9  4 4.000 4.0
# 11 6 5.905 5.0
# 6  2 2.500 2.5
# 8  3 3.500 3.5
```

Two mechanisms combine:

1. The generator rounds the centre to an integer with `torch.round(width * center)` (`kornia/augmentation/random_generator/_2d/gaussian_illumination.py`). `torch.round` rounds half to even, so `0.5 * 7 = 3.5` goes to `4` while `0.5 * 9 = 4.5` goes to `4`. On odd axes the result alternates between the centre and one pixel past it.
2. `kornia.filters.kernels.gaussian` adds `0.5` to the sample offsets for an even `window_size` even when an explicit `mean` is passed, so on an even axis the peak sits at `round(center * L) - 0.5`. For `center=0.5` that happens to be the true centre, but at `center=0` or `1` the weight lands on a single edge pixel while interior centres split between two.

The same offsets apply at every `sigma`, including the `sigma=0` limit, where the field is an impulse at `round(center * L)` on an odd axis and splits between `round(center * L) - 1` and `round(center * L)` on an even one.

A fix would map `center` to the pixel-centre coordinate `center * L - 0.5` (or `center * (L - 1)`) without integer rounding, and pass `gaussian` a mean that is not shifted again.

#4793 documents this as a known issue on `RandomGaussianIllumination`.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
